docs: Clean up quick start steps for editing the task (#114)
brown-ccv/honeycomb-docs: docs/quick_start.mdx

## Run the Task

```shell title="Run the task in development mode"
npm run dev
```

> Running the task in development mode causes allows for hot-reloading whenever changes are made to the app. This is how you'll run the project while creating your task!

Reviewer: ```suggestion
Running the task in development mode enables "hot reloading": when you make changes to the code, they'll immediately be reflected in the app without you needing to restart the server. This is how you'll run the project while creating your task!
```

> Running the task in development mode enables "hot reloading": when you make changes to the code, they'll immediately be reflected in the app without you needing to restart the server. This is how you'll run the project while creating your task!

Author: ```suggestion
Running the task in development mode enables "hot reloading": changes to the code will immediately be reflected in the app without needing to restart the server. This is how you'll run the project while creating your task!
```

:::tip
The dev script runs Honeycomb on Electron without any environment variables. Check out the [NPM Scripts](npm_scripts#dev) page for more information on the available development environments.
:::

:::note
Honeycomb ships with a modified version of the "simple reaction time task" from the [jsPsych tutorial](https://www.jspsych.org/7.3/tutorials/rt-task/). In the next section we'll create a new task and tell Honeycomb to run it!
:::

## Edit the Task

Now that the project is up and running we can make our first changes to the code!

### 1) Edit the Project Metadata

> 1. Create a new branch

Reviewer: Do you need to specify that this command is run in the Terminal and without killing the server?

Reviewer: I mean, you could rerun it, but maybe folks need to be reminded that the server needs to be running constantly or else the app will close?

Author: I think it's okay to leave it as is. Either closing the server and changing branches or changing branches in a new terminal will work.  We document that the `npm run dev` will reflect the changes immediately so whether or not they keep it running is up to them

Author: I updated the text to say "Create a new branch in a separate terminal" which should help a little

> 1. Create a new branch

Author: ```suggestion
1. Create a new branch in a separate terminal
```

    *
    * @param {Object} jsPsych The jsPsych instance being used to run the task
    * @returns {Object} A jsPsych timeline object
    */
   export function buildTaskNameTimeline(jsPsych) {}
   ```

7. Save your changes and commit them to git:

   ```shell
   git commit -a -m "feat: adds taskNameOptions and buildTaskNameTimeline to taskName.js"
   ```

8. Edit `src/experiment/index.js` to use the new file

   ```js title="src/experiment/index.js"
   // highlight-delete-next-line
   import { buildHoneycombTimeline, honeycombOptions } from "./honeycomb";
   import { buildTaskNameTimeline, taskNameOptions } from "./taskName";

   // ...

   // highlight-delete-next-line
   export const jsPsychOptions = honeycombOptions;
   export const jsPsychOptions = taskNameOptions;

   // ...

   export function buildTimeline(jsPsych, studyID, participantID) {
     console.log(`Building timeline for participant ${participantID} on study ${studyID}`);

     // highlight-delete-next-line
     const timeline = buildHoneycombTimeline(jsPsych);
     const timeline = buildTaskNameTimeline(jsPsych);
     return timeline;
   }
   ```

> 9. Run a format to make sure the code is formatted correctly

Reviewer: ```suggestion
9. Run the format script to make sure the code is formatted correctly
```

   ```javascript title="taskName.js"
   import { buildStartProcedure } from "./procedures/startProcedure";
   import { buildEndProcedure } from "./procedures/endProcedure";

   // ...

   export function buildTaskNameTimeline(jsPsych) {
     // Build the trials that make up the start procedure
     const startProcedure = buildStartProcedure(jsPsych);

     // Builds the trials that make up the end procedure
     const endProcedure = buildEndProcedure(jsPsych);

     const timeline = [startProcedure, endProcedure];
     return timeline;
   }
   // ...
   ```

   :::tip
   The text for the conclusion trial is in `src/config/language.json` under the `trials` and `conclusion` key.

   ```json title="src/config/language.json"
    {
      "name": "taskName"
      // ...
      "trials": {
        // ...
        "conclusion": "Welcome to the experiment. Press any key to begin."
      },
    }
   ```

   :::

8. Run a format to make sure the code is formatted correctly

   ```shell
   npm run format
   ```

9. Save your changes and commit them to git:

   ```shell
   git commit -a -m "feat: adds endProcedure to the task"
   ```

> 10. Create and merge a [pull request](version_control#create-a-pull-request) to merge your changes into the `main` branch. Make sure the builds complete successfully before merging!

Reviewer: Do you want to give instructions on what it means to make sure the build completes successfully?

Author: The builds are run as github actions and I think GitHub makes it pretty self explanatory if/when the actions fail and how to diagnose the issues. I'm toying around with the idea of writing a seperate GitHub Discussions post for working with the actions but I don't think that needs to be up before 3.3 gets released
